robot/Cumulus/tests/browser/crlps/total_current_year_payments.robot

```diff
@@ -0,0 +1,96 @@
+    Set suite variable                @{filterPools}
+
+
+
+*** Variables ***
+&{contact1_fields}       Email=test@example.com
+
+*** Test Cases ***
+
+Total Current Year Payments on Prior Year Pledges
+    [Documentation]       Calculates Total Current Year Payments on Prior Year opportunity Pledges
+    ...                   Enables CRLP settings, creates the custom fields on account and payment objects required for rollup
+    ...                   Creates a filter and crlp setting to rollup the amount of current year payments on a previous year opportunity
+
+    [tags]                            unstable                          W-037650                           feature:crlps
+
+    # Create a Filter group and CRLP setting after checking prior records do not check if element exists
+    Load Page Object                                      Custom                          CustomRollupSettings
+    Navigate To Crlpsettings
+
+    Create New Filter Setting
+    ...                                                   @{filterPools}  # Contains a list of filter metadata
+    ...                                                   &{dict}         # Contains the name of the fitler and the description to be added
+
+    Click Link With Text                                  Customizable Rollups
+    Create New Rollup Setting
+        ...                                                Target Object=Account
+        ...                                                Target Field=This Year Payments on Past Year Pledges
+        ...                                                Description=Shows Payments made this year for pledges (Opportunities) from the prior year.
+
+        ...                                                Operation=Sum
+        ...                                                Time Frame=Years Ago
+        ...                                                Years Ago=This Year
+        ...                                                Rollup Type=Payment -> Account (Hard Credit)
+
+        ...                                                Filter Group=Old Payments
+        ...                                                Date Field=Payment: Payment Date
+
+    # Create an opportunity for prior year that has 6 payments associated with a contact .
+    # Ensure that Two of the payments were done the current year and one is paid last year.
+    ${opp_date} =            Get Current Date    result_format=%Y-%m-%d    increment=-365 days
+    ${payment_fields} =      Create Dictionary   Amount=833.34   NumPayments=6   Scheduledate=${opp_date}      Interval=1   CompletedPayments=3
+    ${opportunity_fields} =  Create Dictionary   Type=Donation
+    ...                                          Name=Auto Payment test $1000 Donation
+    ...                                          Amount=5000     CloseDate=${opp_date}    StageName=Closed Won    npe01__Do_Not_Automatically_Create_Payment__c=true
```

Comment:
Can u move these dictionary creations to setup as well

Reply:
This one I am not comfortable moving as I have to create this opportunity only after setting up the filters and crlp setting. The readability of the test is gone as I am documenting how I am generating the opportunity for prior year .